### PR TITLE
Add DPUCAHX8L target

### PR DESCRIPTION
**Pull request by anilmartha**

### CODE REVIEW

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        bank_depth=4096,
+        max_pool_min_kernel=2,
+        max_pool_max_kernel=8,
+        max_pool_min_stride=1,
+        max_pool_max_stride=8,
+        avg_pool_min_kernel=1,
+        avg_pool_max_kernel=8,
+        avg_pool_min_stride=1,
+        avg_pool_max_stride=8,
+        max_pool_kernel_valid=[ 2, 3, 5, 7, 8],
```
Comment (jtuyls) [inline python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py:145]: max_pool_kernel_valid isn't necessary for this DPU: https://www.xilinx.com/html_docs/vitis_ai/1_3/compiling_model.html#ztl1570696058091

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+def mean_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """ Check whether we can execute the provided Mean operator
+        on the DPUCAHX8L-u280 target """
+    axes = X.attrs["axes"]
+    keepdims = X.attrs["keepdims"]
+    return len(axes) == 2 and keepdims
+
+
+@pyxir.register_op_support_check("DPUCAHX8L-u280", "LeakyReLU")
+def leaky_relu_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
```
Comment (jtuyls) [inline python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py:160]: Leaky relu isn't supported for this DPU: https://www.xilinx.com/html_docs/vitis_ai/1_3/compiling_model.html#ztl1570696058091

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+    return math.isclose(alpha, 0.1, rel_tol=1e-5) and bX.type[0] in [
+        "Convolution",
+        "Conv2DTranspose",
+        "BatchNorm",
+        "Scale",
+    ]
+
+
+@pyxir.register_op_support_check("DPUCAHX8L-u280", "pReLU")
+def prelu_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
```
Comment (jtuyls) [inline python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py:176]: Same for prelu as leaky relu above

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+    # TODO always?
+
+    return True
+
+
+@pyxir.register_op_support_check("DPUCAHX8L-u280", "ReLU6")
+def relu6_op_support(X: XLayer, bXs: List[XLayer], tXs: List[XLayer]) -> bool:
+    """ Check whether we can execute the provided ReLU6 operator
+        on the DPUCAHX8L-u280 target """
+    return False
```
Comment (jtuyls) [inline python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py:207]: Relu6 is supported for DPU after Convolution, Transposed Conv2D, BatchNorm, Scale, Eltwise

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+        on the DPUCAHX8L-u280 target """
+    return is_conv2d_supported(
+        X,
+        bXs,
+        tXs,
+        channel_parallel=32,
+        bank_depth=4096,
+        max_stride=4,
+        min_stride=1,
+        depthwise_supported=False,
```
Comment (jtuyls) [inline python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py:82]: Depthwise Conv2D is supported

File: tests/unit/contrib/test_dpucahx8l.py [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        import pyxir.contrib.target.DPUCAHX8L
+
+    @classmethod
+    def tearDownClass(cls):
+        # Unregister dpu for other tests
+        TestDPUCAHX8H.target_registry.unregister_target("DPUCAHX8L-u50")
+        TestDPUCAHX8H.target_registry.unregister_target("DPUCAHX8L-u280")
+
+    def test_compile_conv2d_pool2d(self):
+        xcompiler_conv2d_pool2d_nhwc_oihw_test(
```
Comment (jtuyls) [inline tests/unit/contrib/test_dpucahx8l.py:60]: You will have to add the targets here, default is DPUCAHX8H

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py (hunk 3 above)
Comment (anilmartha) [reply python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py:166]: added

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py (hunk 5 above)
Comment (anilmartha) [reply python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py]: added

File: tests/unit/contrib/test_dpucahx8l.py (hunk 6 above)
Comment (anilmartha) [reply tests/unit/contrib/test_dpucahx8l.py:60]: Added targets

File: python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py (hunk 4 above)
Comment (anilmartha) [reply python/pyxir/contrib/target/components/DPUCAHX8L/u280_op_support.py]: added

**Pull request merged**